src/solidlsp/language_servers/csharp_language_server.py

```
@@ -204,266 +211,276 @@ def __init__(self, config: LanguageServerConfig, repository_root_path: str, soli
+            return dotnet_path, server_dll_path
+
+        def _ensure_dotnet_runtime(self, dotnet_runtime_dep: RuntimeDependency) -> str:
+            """Ensure .NET runtime is available and return the dotnet executable path."""
+            # TODO: use RuntimeDependency util methods instead of custom validation/download logic
+
+            # Check if dotnet is already available on the system
+            system_dotnet = shutil.which("dotnet")
+            if system_dotnet:
+                # Check if it's .NET 9
+                try:
+                    result = subprocess.run([system_dotnet, "--list-runtimes"], capture_output=True, text=True, check=True)
```

Ruff (0.14.11)

`subprocess` call: check for execution of untrusted input

(S603)